lib/rubygems/commands/setup_command.rb

```diff
@@ -400,16 +400,18 @@ def install_default_bundler_gem(bin_dir)
     Dir.chdir("bundler") do
       built_gem = Gem::Package.build(new_bundler_spec)
       begin
-        Gem::Installer.at(
+        installer = Gem::Installer.at(
           built_gem,
           env_shebang: options[:env_shebang],
           format_executable: options[:format_executable],
           force: options[:force],
-          install_as_default: true,
           bin_dir: bin_dir,
           install_dir: default_dir,
           wrappers: true
-        ).install
+        )
+        installer.install
+        File.delete installer.spec_file
+        installer.write_default_spec
```

Comment:
I think this will install a full copy of bundler in the standard `$GEM_HOME`. I don't think it's a problem per se, but it makes Bundler inconsistent with other default gems with executables provided with Ruby (for example, irb, which includes a placeholder gem directory with just the executable).

We could try to iterate on this but perhaps for now it's better to not use `Gem::Installer` at all here and create the files necessary (executables, and gemspec) manually, to keep the exact same folder structure `gem update --system` creates now.